olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp

```diff
@@ -155,40 +155,44 @@ ApiClientLookup::ApiClientResponse ApiClientLookup::LookupApi(
   auto client = std::make_shared<client::OlpClient>();
   client->SetSettings(std::move(settings));
 
-  client::Condition condition(cancellation_context);
+  client::Condition condition(
+      std::chrono::seconds(settings.retry_settings.timeout));
 
-  // when the network operation took too much time we cancel it and exit
-  // execution, to make sure that network callback will not access dangling
-  // references we protect them with atomic bool flag.
-  auto interest_flag = std::make_shared<std::atomic_bool>(true);
+  auto flag = std::make_shared<std::atomic_bool>(true);
 
   ApiClientLookup::ApiClientResponse api_response;
+  auto apiClientCallback =
+      [&](ApiClientLookup::ApiClientResponse response) {
+        if (flag->exchange(false)) {
+          api_response = std::move(response);
+          condition.Notify();
+        }
+      };
+
   cancellation_context.ExecuteOrCancelled([&]() {
-    return ApiClientLookup::LookupApiClient(
-        client, service, service_version, catalog,
-        [&,
-         interest_flag](ApiClientLookup::ApiClientResponse response) mutable {
-          if (interest_flag->exchange(false)) {
-            api_response = std::move(response);
-            condition.Notify();
-          }
-        });
+    auto token = ApiClientLookup::LookupApiClient(
+        client, service, service_version, catalog, apiClientCallback);
```

Comment:
```suggestion
        client, service, service_version, catalog, std::move(apiClientCallback));
```
Please, correct me if I'm wrong here :)

Reply:
This is part of different pull request. My change was rebased on it.